radicale/storage.py

```diff
@@ -304,23 +357,40 @@ def get(self, href):
         raise NotImplementedError
 
     def get_multi(self, hrefs):
-        """Fetch multiple items. Duplicate hrefs must be ignored.
+        """Fetch multiple items.
+
+        Functionally similar to ``get``, but might bring performance benefits
+        on some storages when used cleverly. It's not required to return the
+        requested items in the correct order. Duplicated hrefs can be ignored.
+
+        Returns tuples with the href and the item or None if the item doesn't
+        exist.
+
+        """
+        return map(lambda href: (href, self.get(href)), hrefs)
```

Comment:
No, that would defeat the main purpose of this PR. A generator expression will load everything into memory at once. Avoiding generator expressions is the main reason, why this PR reduces the memory usage significantly.